Short-Answer/Algorithms_Answers.md

```diff
@@ -3,14 +3,25 @@
 ## Exercise I
 
 a)
-    If n > 0 input size n is going to always be bigger than a by n. So at worst it'll loop over one more time and it'll do this constantly no matter how large the input which is O(1) time complexity.
+    If n > 0 input size n is going to always be bigger than a by *n so with a really big input its the operations are going to increase linearly so it can be greater than n^3. n^3/n^2 = n, O(n) time complexity.
 
 b)
-    The first loop is O(n) because each operation increases linearly based on input size. The second, while loop, is also O(n) for the same reason so in total this is O(n^2) time complexity.
+    The first loop is O(n) because each operation increases linearly based on input size. The second, while loop, is O(log(n)) because it talks half of the operations to run. For this reason its total this is O(n log(n)) time complexity.
 
 c)
     The function is being called recursively n times before reaching the base case so its O(n).
 
 ## Exercise II
 
+while n > 1:
+  start at middle building
+  drop egg
+  if egg is broken:
+    remove all values > n
+  else:
+    remove all values =< n
+  n/2,repeat above code
 
```

Comment:
I understand what this is doing just think of ways to improve your pseudo code (it doesn't have to look like code at all) i.e. If egg breaks then eliminate all floors above current floor and halve the remaining floors below me. Drop another egg from the middle floor of the remaining floors...rinse and repeat...